### PR TITLE
Feature/split app building (#43)

### PR DESCRIPTION
**Pull request by michal367**
Separation of building the lecturer and student application

### CODE REVIEW

File: server/server.ts [hunk 1]
```diff
@@ -3,19 +3,19 @@ import * as path from "https://deno.land/std/path/mod.ts";
 import router from "./routes.ts";
 
 const PORT = 8000;
-const HOST = "localhost";
+//const HOST = "localhost";
```
Comment (ostatni5) [inline server/server.ts:6]: This comment is necessary?

**Review by Derstilon — CHANGES_REQUESTED** [review]
image
Czemu dwa razy ten plik dodałeś App.tsx do lecturer?
[ta wersja](https://github.com/michal367/OI-Project/blob/1eb11ca5d023bd27e9b545a739b51d138bc3f464/src/lecturer/App.tsx) jest zbędna

**Review by ostatni5 — APPROVED** [review]
I only don't understand the necessity of changes done in server/server.ts.
The rest seems fine to me, apps build and start correctly.

File: src/student/reportWebVitals.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from 'web-vitals';
+
+const reportWebVitals = (onPerfEntry?: ReportHandler) => {
+  if (onPerfEntry && onPerfEntry instanceof Function) {
+    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
+      getCLS(onPerfEntry);
+      getFID(onPerfEntry);
+      getFCP(onPerfEntry);
```
Comment (Petros9) [inline src/student/reportWebVitals.ts:8]: What are these functions for? (getters)

File: server/server.ts (hunk 1 above)
Comment (michal367) [reply server/server.ts:6]: no, but before my change it was used to start server, but it only started server on 127.0.0.1 and not on local network. After change it starts on both. So I can delete this comment.

File: src/student/serviceWorkerRegistration.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/)
+);
+
+type Config = {
+  onSuccess?: (registration: ServiceWorkerRegistration) => void;
+  onUpdate?: (registration: ServiceWorkerRegistration) => void;
+};
+
+export function register(config?: Config) {
+  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
```
Comment (Petros9) [inline src/student/serviceWorkerRegistration.ts:27]: is navigator somehow imported or declared before? what are his responsibilities?

File: src/student/reportWebVitals.ts (hunk 2 above)
Comment (michal367) [reply src/student/reportWebVitals.ts:8]: It's not my file. It is from standard react app template. Student app for now is just template.

File: src/student/serviceWorkerRegistration.ts (hunk 3 above)
Comment (michal367) [reply src/student/serviceWorkerRegistration.ts:27]: It's not my file. It is from PWA react app template. Student app for now is just PWA react template. This file is used by PWA.

File: src/student/reportWebVitals.ts (hunk 2 above)
Comment (Petros9) [reply src/student/reportWebVitals.ts:8]: alright

**Review by Petros9 — APPROVED** [review]

**Review by Derstilon — APPROVED** [review]
I have no further objections.

**Pull request merged**